### PR TITLE
docs(explore/dpp): rework dpp page

### PR DESCRIPTION
**Pull request by lahabana**
Signed-off-by: Charly Molter <charly.molter@konghq.com>

### CODE REVIEW

File: docs/docs/dev/explore/dpp.md [hunk 1]
```diff
@@ -1,64 +1,86 @@
...
-## Dataplane Entity
+## Concepts: Inbounds, Outbounds, Tags and Services
 
-A `Dataplane` entity must be passed to `kuma-dp` when instances attempt to connect to the control plane.
-On Kubernetes, this operation is [fully **automated**](dpp-on-kubernetes.md).
-On Universal, it must be executed [**manually**](dpp-on-universal.md).
+### Inbound
 
-To understand why the `Dataplane` entity is required, we must take a step back. As we have explained already, Kuma follows a sidecar proxy model for the data plane proxies, where we have an instance of a data plane proxy for every instance of our services. Each Service and DP will communicate with each other on the same machine, therefore on `127.0.0.1`.
+Inbounds are defined in the `Dataplane` entity and is a combination of a port and a set of tags.
```
Comment (lukidzi) [inline docs/docs/dev/explore/dpp.md:26]: Is it correct? 
```suggestion
Inbounds are defined in the `Dataplane` entity and are a combination of a port and a set of tags.
```
Shouldn't outbounds be moved after Inbounds to make the same order as in the title?

File: docs/docs/dev/explore/dpp.md [hunk 2]
```diff
@@ -1,64 +1,86 @@
...
+The port is the local port that the workload binds to for this inbound.
 
-For example, if we have 6 replicas of a "Redis" service, then we must have one instances of `kuma-dp` running alongside each replica of the service, therefore 6 replicas of `kuma-dp` and 6 `Dataplane` entities as well.
+Most of the time a DPP exposes a single inbound. When a workload exposes multiple ports, multiple inbounds can be defined.
 
-<center>
-<img src="/images/docs/0.4.0/diagram-11.jpg" alt="" style="width: 500px; padding-top: 20px; padding-bottom: 10px;"/>
-</center>
+### Tags
+Tags are a set of key-values that are defined for each DPP inbound. These tags serve these purposes:
```
Comment (lukidzi) [inline docs/docs/dev/explore/dpp.md:32]:
```suggestion
Tags are a set of key-values that are defined for each DPP inbound. These tags serve the following purposes:
```

Comment (parkanzky) [reply docs/docs/dev/explore/dpp.md:32]: "key-value pairs" ?

File: docs/docs/dev/explore/dpp.md [hunk 3]
```diff
@@ -1,64 +1,86 @@
...
 
-::: tip
-**Many DPs!** The number of data plane proxies that we have running can quickly add up, since we have one replica of `kuma-dp` for every replica of every service. That's why it's important for the `kuma-dp` process to be lightweight and consume few resources, otherwise we would quickly run out of memory, especially on platforms like Kubernetes where multiple services are running on the same underlying host machine. And that's one of the reasons Kuma leverages Envoy for this task.
-:::
+- Define which service this DPP inbound is part of.
+- Add some metadata on the nature of the service being exposed. 
+- Be able to select subsets of dataplanes using these tags.
```
Comment (parkanzky) [inline docs/docs/dev/explore/dpp.md:36]: Allow subsets of data planes to be selected by these tags.

File: docs/docs/dev/explore/dpp.md [hunk 4]
```diff
@@ -1,64 +1,86 @@
 # Data plane proxy
 
-When Kuma (`kuma-cp`) runs, it waits for the data plane proxies to connect and register themselves.
+A **data plane proxy (DPP)** is the part of Kuma that runs next to each workload that is a member of the mesh.
+A DPP is composed of the following components:
+
+- a `Dataplane` entity which defines the configuration of the data plane proxy.
```
Comment (lahabana) [inline docs/docs/dev/explore/dpp.md:6]:
```suggestion
- a `Dataplane` entity defines the configuration of the DPP
```

File: docs/docs/dev/explore/dpp.md [hunk 5]
```diff
@@ -1,64 +1,86 @@
...
+- a `kuma-dp` binary that runs on each instance that is part of the mesh. This binary spawns the following subprocesses:
```
Comment (lahabana) [inline docs/docs/dev/explore/dpp.md:7]:
```suggestion
- a `kuma-dp` binary runs on each instance that is part of the mesh. This binary spawns the following subprocesses:
```

File: docs/docs/dev/explore/dpp.md [hunk 6]
```diff
@@ -1,64 +1,86 @@
...
+  - `envoy` which will receive configuration from the control-plane to manage traffic correctly
```
Comment (lahabana) [inline docs/docs/dev/explore/dpp.md:8]:
```suggestion
  - `Envoy` receives configuration from the control-plane to manage traffic correctly 
```

File: docs/docs/dev/explore/dpp.md [hunk 7]
```diff
@@ -1,64 +1,86 @@
 # Data plane proxy
 
-When Kuma (`kuma-cp`) runs, it waits for the data plane proxies to connect and register themselves.
+A **data plane proxy (DPP)** is the part of Kuma that runs next to each workload that is a member of the mesh.
+A DPP is composed of the following components:
+
+- a `Dataplane` entity which defines the configuration of the data plane proxy.
+- a `kuma-dp` binary that runs on each instance that is part of the mesh. This binary spawns the following subprocesses:
+  - `envoy` which will receive configuration from the control-plane to manage traffic correctly 
+  - `core-dns` which will help resolve Kuma specific DNS entries
```
Comment (lahabana) [inline docs/docs/dev/explore/dpp.md:9]:
```suggestion
  - `core-dns` resolves Kuma specific DNS entries
```

File: docs/docs/dev/explore/dpp.md [hunk 8]
```diff
@@ -1,64 +1,86 @@
...
 
-::: tip
-**Many DPs!** The number of data plane proxies that we have running can quickly add up, since we have one replica of `kuma-dp` for every replica of every service. That's why it's important for the `kuma-dp` process to be lightweight and consume few resources, otherwise we would quickly run out of memory, especially on platforms like Kubernetes where multiple services are running on the same underlying host machine. And that's one of the reasons Kuma leverages Envoy for this task.
-:::
+- Define which service this DPP inbound is part of.
+- Add some metadata on the nature of the service being exposed. 
+- Be able to select subsets of dataplanes using these tags.
 
-When we start a new data plane proxy in Kuma, it needs to communicate a few things to the control-plane: 
+Some tags are reserved to Kuma are prefixed with `kuma.io` like:
```
Comment (parkanzky) [inline docs/docs/dev/explore/dpp.md:38]: Do users generally add their own `kuma.io/*` tags? If not, we might want to more broadly say "Tags prefixed by kuma.io are reserved to Kuma". It's simpler than "some tags are reserved and are prefixed" and may prevent users from creating a tag we might reserve in the future.

File: docs/docs/dev/explore/dpp.md [hunk 9]
```diff
@@ -1,64 +1,86 @@
...
+A **data plane proxy (DPP)** is the part of Kuma that runs next to each workload that is a member of the mesh.
+A DPP is composed of the following components:
+
+- a `Dataplane` entity which defines the configuration of the data plane proxy.
+- a `kuma-dp` binary that runs on each instance that is part of the mesh. This binary spawns the following subprocesses:
+  - `envoy` which will receive configuration from the control-plane to manage traffic correctly 
+  - `core-dns` which will help resolve Kuma specific DNS entries
+
+::: tip
+Data plane proxies are also often called sidecars.
```
Comment (lahabana) [inline docs/docs/dev/explore/dpp.md:12]:
```suggestion
Data plane proxies are also called sidecars.
```

File: docs/docs/dev/explore/dpp.md [hunk 10]
```diff
@@ -1,64 +1,86 @@
...
+:::
+
+Since we have one replica of `kuma-dp` for every replica of every service, the number of DPP running quickly adds up.
```
Comment (lahabana) [inline docs/docs/dev/explore/dpp.md:15]:
```suggestion
We have one instance of `kuma-dp` for every instance of every service.
```

File: docs/docs/dev/explore/dpp.md [hunk 11]
```diff
@@ -1,64 +1,86 @@
...
+- Kubernetes: the `Dataplane` entity is generated [**automatically**](dpp-on-kubernetes.md) by the control-plane.
+- Universal: the `Dataplane` entity must be defined [**manually**](dpp-on-universal.md).
+ 
+The `Dataplane` entity includes the sections that are described below in the [Dataplane Specification](../generated/resources/proxy_dataplane.md):
 
-::: tip
-In order for a data plane proxy to successfully run, there must exist at least one [`Mesh`](../../policies/mesh) in Kuma.
-By default, the system generates a `default` Mesh when the control-plane is run for the first time.
-:::
+* `networking` define what services are exposed, transparent-proxy configuration, inbounds, and outbounds.
```
Comment (parkanzky) [inline docs/docs/dev/explore/dpp.md:67]: defines

Comment (parkanzky) [reply docs/docs/dev/explore/dpp.md:67]: actually "to define", to keep consistent w the rest of the bullets

File: docs/docs/dev/explore/dpp.md [hunk 12]
```diff
@@ -1,64 +1,86 @@
...
-In order for a data plane proxy to successfully run, there must exist at least one [`Mesh`](../../policies/mesh) in Kuma.
-By default, the system generates a `default` Mesh when the control-plane is run for the first time.
-:::
+* `networking` define what services are exposed, transparent-proxy configuration, inbounds, and outbounds. 
+* `metrics` to configure how to collect metrics from the data plane proxy and the workload, check the [traffic metrics](../policies/traffic-metrics.md#expose-metrics-from-applications) for more details.
+* `probes` to configure unsecure port to access workload's health endpoints for platform health probing, check the [health probes](../policies/service-health-probes.md) for more details.
+
+## How data plane proxies get configured
+
+As mentioned previously each DPP has a `Dataplane` entity attached to it which describes what and how a workload should expose.
```
Comment (parkanzky) [inline docs/docs/dev/explore/dpp.md:73]: should be exposed

File: docs/docs/dev/explore/dpp.md [hunk 13]
```diff
@@ -1,64 +1,86 @@
...
+- a `kuma-dp` binary that runs on each instance that is part of the mesh. This binary spawns the following subprocesses:
+  - `envoy` which will receive configuration from the control-plane to manage traffic correctly 
+  - `core-dns` which will help resolve Kuma specific DNS entries
+
+::: tip
+Data plane proxies are also often called sidecars.
+:::
+
+Since we have one replica of `kuma-dp` for every replica of every service, the number of DPP running quickly adds up. 
+For example, if we have 6 replicas of a "Redis" service, then we must have one instance of `kuma-dp` running alongside each replica of the service, therefore 6 replicas of `kuma-dp` and 6 `Dataplane` entities as well.
```
Comment (lahabana) [inline docs/docs/dev/explore/dpp.md:16]:
```suggestion

```

File: docs/docs/dev/explore/dpp.md [hunk 14]
```diff
@@ -1,64 +1,86 @@
...
 
 <center>
-<img src="/images/docs/0.4.0/diagram-10.jpg" alt="" style="width: 500px; padding-top: 20px; padding-bottom: 10px;"/>
+<img src="/images/docs/0.4.0/diagram-11.jpg" alt="" style="width: 500px; padding-top: 20px; padding-bottom: 10px;"/>
 </center>
 
-## Dataplane Entity
+## Concepts: Inbounds, Outbounds, Tags and Services
```
Comment (lahabana) [inline docs/docs/dev/explore/dpp.md:22]:
```suggestion
## Concepts
```

File: docs/docs/dev/explore/dpp.md (hunk 3 above)
Comment (lahabana) [inline docs/docs/dev/explore/dpp.md:36]:
```suggestion
- Allow subsets of data planes to be selected by these tags.
```

File: docs/docs/dev/explore/dpp.md [hunk 15]
```diff
@@ -1,94 +1,78 @@
...
 
-When Kuma (`kuma-cp`) runs, it waits for the data plane proxies to connect and register themselves.
+A **data plane proxy (DPP)** is the part of Kuma that runs next to each workload that is a member of the mesh.
+A DPP is composed of the following components:
 
-<center>
-<img src="/images/docs/0.4.0/diagram-10.jpg" alt="" style="width: 500px; padding-top: 20px; padding-bottom: 10px;"/>
-</center>
+- a `Dataplane` entity defines the configuration of the DPP
+- a `kuma-dp` binary runs on each instance that is part of the mesh. This binary spawns the following subprocesses:
```
Comment (slonka) [inline docs/docs/dev/explore/dpp.md:7]: nitpick: this `a` looks a bit weird and the nested list does not use articles - can we omit it here as well?

File: docs/docs/dev/explore/dpp.md [hunk 16]
```diff
@@ -1,94 +1,78 @@
...
+Most of the time a DPP exposes a single inbound. When a workload exposes multiple ports, multiple inbounds can be defined.
 
-Because these dataplane types are specific and complex we will discuss them separately to "standard" dataplane proxies.
-:::
+### Tags
+Tags are a set of key-value pairs (.e.g `version=v2`) that are defined for each DPP inbound. These tags serve the following purposes:
 
-To do this, we have to create a file with a `Dataplane` definition and pass it to `kuma-dp run`. This way, data-plane will be registered in the Control Plane and Envoy will start accepting requests.
+- specifying the service this DPP inbound is part of
+- adding some metadata about the exposed service
```
Comment (slonka) [inline docs/docs/dev/explore/dpp.md:36]:
```suggestion
- adding metadata about the exposed service
```

**Review by slonka — APPROVED** [review]
Small nitpicks, overall good job 👍

**Pull request merged**